daxa/mission/erosita.py

```diff
@@ -0,0 +1,703 @@
+        :return: The field dictionary.
+        :rtype: dict
+        """
+        # Creating a dictionary to store obs_ids as keys and their field name as values
+        field_dict = {}
+        for field in self._miss_poss_fields:
+            obs = CalPV_info["Obs_ID"].loc[CalPV_info["Field_Name"] == field].values[0]
+            if "," in obs:
+                # This checks if multiple observations are associated with one field
+                indv_obs = obs.split(", ")
+                for ind_ob in indv_obs:
+                    field_dict[ind_ob] = field
+            else:
+                field_dict[obs] = field
+        
+        return field_dict
+
+    
+    @_lock_check
+    def filter_on_fields(self, fields: Union[str, List[str]]):
+        """
+        This filtering method will select only observations included in the fields specified.
+
+        :param str/List[str] allowed_fields: The fields or field types (or list of fields or field types) 
+            that you wish to be let through the filter.
+        """
+
+        # Convert field types or a singular field name into a list of field name(s)
+        fields = self._check_chos_fields(fields=fields)
+
+        # Updating the chosen_field attribute
+        self.chosen_fields = fields
+
+        # Creating a dictionary that store obs_ids as keys and their field name as values
+        field_dict = self._field_dict_generator()
+        
+        # Selecting all Obs_IDs from each field
+        field_obs_ids = [obs for field in fields for obs in field_dict if field_dict[obs] == field]
+
+        # Uses the Pandas isin functionality to find the rows of the overall observation table that match the input
+        #  ObsIDs. This outputs a boolean array.
+        sel_obs_mask = self._obs_info['ObsID'].isin(field_obs_ids).values
+        # Said boolean array can be multiplied with the existing filter array (by default all ones, which means
+        #  all observations are let through) to produce an updated filter.
+        new_filter = self.filter_array*sel_obs_mask
+        # Then we set the filter array property with that updated mask
+        self.filter_array = new_filter
+    
+    # Then define user-facing methods
+    def fetch_obs_info(self):
+        """
+        This method uses the hard coded csv file to pull information on all eROSITACalPV observations. 
+        The data are processed into a Pandas dataframe and stored.
+        """
+        # Hard coded this and saved it to the obs_info.csv in /files
+        obs_info['ObsID'] = [str(obs) for obs in obs_info['ObsID']]
+        obs_info['start'] = [time.split('.', 1)[0] for time in obs_info['start']]
+        obs_info['end'] = [time.split('.', 1)[0] for time in obs_info['end']]
+        obs_info['start'] = pd.to_datetime(obs_info['start'], utc=False, format="%Y-%m-%dT%H:%M:%S", errors='coerce')
+        obs_info['end'] = pd.to_datetime(obs_info['end'], utc=False, format="%Y-%m-%dT%H:%M:%S", errors='coerce')
+
```

Comment:
Before you send it to the all_obs_info property it would be nice to re-order the columns. One way of doing it is this:

https://github.com/DavidT3/DAXA/blob/a04557d0ba2b2ba5eaf7da56b7b2418cfb714ff2/daxa/mission/nustar.py#L278-L281

Though if you forget to put a column in the list of column names it will be removed from the dataframe so just be wary of that. 

I just think it's nicer to have the ra, dec, and ObsID being the first things the user sees on the left - entirely aesthetic decision though

Reply:
no worries, I've put this in the updated version of _fetch_obs_info() when I merged the two csv's